### ISSUE TITLE
Investigate issues with swapping aghs scepter in and out of inventory

### ISSUE THREAD

**Issue by Draze22** [arcadia-redux/custom_hero_clash_issues#2201]
![image](https://user-images.githubusercontent.com/63279022/178425548-48610b80-3f4e-462d-b8b5-876bfcb097d4.png)

**Comment by Draze22** [comment]
Couldn't find a way to reproduce this:

![image](https://user-images.githubusercontent.com/63279022/178443779-9f9fe6b1-83dc-4e79-bbc6-87549ec4414d.png)

**Comment by lovro-m** [comment]
Yeah OnScepterReceived() procs before scepter_lost event dispatches in OnScepterLost(). Not sure how to fix this :/ Any ideas?

**Comment by SanctusAnimus** [comment]
Check if player still has scepter in `OnScepterLost` and cancel removal entirely if that's the case
